File: lib/models/prayer.dart
class Prayer {
  final String id;
  final String title;
  final String category;
  final String language;
  final String lyrics;
  final String meaning;
  final int durationMinutes;
  final String audioUrl;
  final String deity;
  final String imageUrl;

  Prayer({
    required this.id,
    required this.title,
    required this.category,
    required this.language,
    required this.lyrics,
    required this.meaning,
    required this.durationMinutes,
    required this.audioUrl,
    required this.deity,
    required this.imageUrl,
  });

  factory Prayer.fromJson(Map<String, dynamic> json) {
    return Prayer(
      id: json['id'] ?? json['_id'] ?? '',
      title: json['title'] ?? '',
      category: json['category'] ?? '',
      language: json['language'] ?? 'Sanskrit',
      lyrics: json['lyrics'] ?? '',
      meaning: json['meaning'] ?? '',
      durationMinutes: json['duration_minutes'] ?? 5,
      audioUrl: json['audio_url'] ?? '',
      deity: json['deity'] ?? '',
      imageUrl: json['image_url'] ?? '',
    );
  }
}

File: backend/models/Prayer.js
const mongoose = require('mongoose');

const prayerSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true,
  },
  category: {
    type: String,
    enum: ['Morning', 'Evening', 'Mantra', 'Other'],
    default: 'Other',
  },
  language: {
    type: String,
    default: 'Sanskrit',
  },
  lyrics: {
    type: String,
    default: '',
  },
  meaning: {
    type: String,
    default: '',
  },
  durationMinutes: {
    type: Number,
    default: 5,
  },
  audioUrl: {
    type: String,
    default: '',
  },
  deity: {
    type: String,
    default: '',
  },
  imageUrl: {
    type: String,
    default: '',
  },
}, { timestamps: true });

module.exports = mongoose.model('Prayer', prayerSchema);